packages/sourcecred/src/core/ledger/ledger.js

```diff
@@ -847,6 +869,153 @@ export class Ledger {
     return this._grainIntegrationStatuses.get(id);
   }
 
+  /**
+   * Disabling accounting has the following effects on the ledger:
+   * - enables grain transfers
+   * - all accounts will start with zero balance again, and it will accumulate
+   *   normally from distributions
+   */
+  enableAccounting(): Ledger {
+    if (!this._balanceAccountingEnabled) {
+      this._createAndProcessEvent({
+        type: "ENABLE_ACCOUNTING",
+      });
+    }
+    // no-op if already enabled
+    return this;
+  }
+
+  _enableAccounting(_: EnableAccounting) {
+    this._balanceAccountingEnabled = true;
+  }
+
+  /**
+   * Disabling accounting has the following effects on the ledger:
+   * - disables grain transfers
+   * - zeroes out all balances
+   * - deactivates accounts that don't have a PayoutAddress set for the
+   *   current instance configuration, and prevents them from activating until
+   *   they do.
+   *
+   *   This function can be called while accounting is disabled to update
+   *   the ledger's external currency configuration. Note that this will also
+   *   deactivate identities missing a payout address that matches the new
+   *   configuration.
+   */
+  disableAccounting(): Ledger {
+    // ensure an external currency is set
+    this.externalCurrency();
+    // ensure accounting isn't already disabled
+    if (this._balanceAccountingEnabled) {
+      this._createAndProcessEvent({
+        type: "DISABLE_ACCOUNTING",
+      });
+    }
+    // no-op if already disabled
+    return this;
+  }
+
+  _disableAccounting(_: DisableAccounting) {
+    this._balanceAccountingEnabled = false;
+
+    const currencyKey = getCurrencyKey(this.externalCurrency());
+
+    // Warning! Mutations below
+    const accountsIterator = this._accounts.values();
+    for (const account of accountsIterator) {
+      // zero out all balances, so there is no confusion about them.
+      // By zeroing them out, they cannot be compared to external balances
+      // in any meaningful way.
+      account.balance = G.ZERO;
+      // deactivate accounts that don't have a payout address assigned
+      // recursive operations are occuring here. Need to be careful with
+      // these next op calls
+      const payoutAddress = account.payoutAddresses.get(currencyKey);
+      if (!payoutAddress) this.deactivate(account.identity.id);
+    }
+  }
+
+  /**
+   * Disabling accounting has the following effects on the ledger:
+   * - disables grain transfers
+   * - zeroes out all balances
+   * - deactivates accounts that don't have a PayoutAddress set for the
+   *   current instance configuration, and prevents them from activating until
+   *   they do.
```

Comment:
This belongs somewhere else or should be removed. It doesn't help me understand setExternalCurrency